src/initialization/InitDistribution.cpp

```diff
@@ -246,6 +248,21 @@ namespace impactx
                 amrex::ParticleReal muypy = distribution.m_muypy;
                 amrex::ParticleReal mutpt = distribution.m_mutpt;
 
+                // some things we cannot represent in envelope mode
+                if (distribution.m_meanx  != 0.0_prt ||
+                    distribution.m_meany  != 0.0_prt ||
+                    distribution.m_meant  != 0.0_prt ||
+                    distribution.m_meanpx != 0.0_prt ||
+                    distribution.m_meanpy != 0.0_prt ||
+                    distribution.m_meanpt != 0.0_prt ||
+                    distribution.m_dispx  != 0.0_prt ||
+                    distribution.m_disppx != 0.0_prt ||
+                    distribution.m_dispy  != 0.0_prt ||
+                    distribution.m_disppy != 0.0_prt
+                ) {
+                    throw std::runtime_error("Cannot create envelope for distribution with non-zero means or dispersion!");
```

Comment:
@cemitch99 If you like, we could open an issue for tracking purposes (does not need to be addressed in the near-term) and we can link the issue for users here.
```suggestion
                    throw std::runtime_error("Cannot (yet) create envelope for distribution with non-zero means or dispersion! Please see: https://github.com/BLAST-ImpactX/impactx/issues/...");
```